src/resolve/resolved-bus.c

```diff
@@ -1580,6 +1582,132 @@ static int bus_method_reset_server_features(sd_bus_message *message, void *userd
         return sd_bus_reply_method_return(message, NULL);
 }
 
+static int bus_method_register_service(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(dnssd_service_freep) DnssdService *service = NULL;
+        _cleanup_free_ char *path = NULL;
+        _cleanup_free_ char *instance_name = NULL;
+        Manager *m = userdata;
+        DnssdService *s = NULL;
+        DnsTxtItem *last = NULL;
+        int r;
+
+        assert(message);
+        assert(m);
+
+        if (m->mdns_support != RESOLVE_SUPPORT_YES)
```

Comment:
this shoul really have some priv checking: either full polkit (permitting unpriv clients by default), or for now require admin privs from the client